common/beerocks/bcl/include/bcl/network/interface_state_monitor.h

```diff
@@ -0,0 +1,86 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef BCL_NETWORK_INTERFACE_STATE_MONITOR_H_
+#define BCL_NETWORK_INTERFACE_STATE_MONITOR_H_
+
+#include <functional>
+#include <string>
+
+namespace beerocks {
+namespace net {
+
+class InterfaceStateMonitor {
+public:
+    /**
+     * Network interface state-change handler function.
+     *
+     * @param iface_name Interface name.
+     * @param iface_state Interface state (true if it is up and running).
+     */
+    using StateChangeHandler = std::function<void(const std::string &iface_name, bool iface_state)>;
```

Comment:
Why the same function for both states? Why not `on_iface_up()` and `on_iface_down()`? Would they share too much code?

Reply:
Both solutions are equivalent, but I think mine is more concise: just one handler type definition, just one setter method to register it and just one member variable to store it. From the caller's perspective it's also simpler (despite it needs an `if` in the handler function to check what's the new interface state)